auth/service.go

```diff
@@ -1064,3 +1069,67 @@ func (svc service) DeleteEntityPolicies(ctx context.Context, entityType, id stri
 		return errInvalidEntityType
 	}
 }
+
+func (svc service) VerifyConnections(ctx context.Context, thingsId, channelsId []string) (ConnectionsPage, error) {
+	uniqueThings := getUniqueValues(thingsId)
+	uniqueChannels := getUniqueValues(channelsId)
+
+	cp := ConnectionsPage{}
+	totalConnectionsCount := len(uniqueChannels) * len(uniqueThings)
+	totalConnectedCount := 0
+
+	cp.Connections = make([]ConnectionStatus, 0, totalConnectionsCount)
+
+	for _, th := range uniqueThings {
+		for _, ch := range uniqueChannels {
+			err := svc.agent.CheckPolicy(ctx, PolicyReq{
+				Subject:     ch,
+				SubjectType: GroupType,
+				Permission:  GroupRelation,
+				Object:      th,
+				ObjectType:  ThingType,
+			})
+			var status string
+			switch {
+			case err == nil:
+				status = connected
+				totalConnectedCount++
+			case errors.Contains(err, svcerr.ErrAuthorization):
+				status = disconnected
+			default:
+				return ConnectionsPage{}, errors.Wrap(svcerr.ErrMalformedEntity, err)
+			}
+
+			cp.Connections = append(cp.Connections, ConnectionStatus{
+				ThingId:   th,
+				ChannelId: ch,
+				Status:    status,
+			})
+		}
+	}
+
+	switch {
+	case totalConnectedCount == totalConnectionsCount:
+		cp.Status = allConn
+	case totalConnectedCount == 0:
+		cp.Status = allDisConn
+	default:
+		cp.Status = partConn
+	}
+
+	return cp, nil
+}
```

Comment:
We can have this logic in Things service,
Like things service will verify connection via auth gRPC connection.

This may be inefficent if we move this logic to things service, but in future we can have direct grpc connection to spiceDB from things service.

Reply:
I have a question concerning this, then how is bootstrap supposed to verify connection without the use of a token?